elpis/gui/src/components/Transcription/ChooseModel.js

```diff
@@ -2,20 +2,37 @@ import React, {Component} from "react";
 import {Link, withRouter} from "react-router-dom";
 import {connect} from "react-redux";
 import {withTranslation} from "react-i18next";
-import {Button, Divider, Grid, Header, Segment} from "semantic-ui-react";
-import {modelLoad, modelList} from "redux/actions/modelActions";
+import {Button, Divider, Grid, Header, Segment, Loader} from "semantic-ui-react";
+import {modelLoad, modelList, modelUpload, modelUploadStatusReset} from "redux/actions/modelActions";
 import {datasetLoad} from "redux/actions/datasetActions";
 import {engineLoad} from "redux/actions/engineActions";
 import {pronDictLoad} from "redux/actions/pronDictActions";
 import Branding from "components/Shared/Branding";
 import urls from "urls";
-
+import Dropzone from "react-dropzone";
+import {fromEvent} from "file-selector";
+import classNames from "classnames";
 
 class ChooseModel extends Component {
     componentDidMount() {
+        if (this.props.uploadStatus === "finished") {
+            this.props._modelUploadFinished();
+            this.props.history.push(urls.gui.transcription.new);
+        }
+
         this.props._modelList();
     }
 
+    componentDidUpdate() {
+        console.log("componentDidUpdate outside if statement");
```

Comment:
We can probably remove these console.log calls now, and maybe refactor the redirect block out into its own function?